plugins/eventMacro/eventMacro/Condition/InChatRoom.pm

```diff
@@ -0,0 +1,34 @@
+package eventMacro::Condition::InChatRoom;
+
+use strict;
+
+use base 'eventMacro::Condition';
+
+use Globals;
+
+sub _hooks {
+	['packet_mapChange','chat_created','chat_leave','chat_joined'];
+}
+
+sub _parse_syntax {
+	my ( $self, $condition_code ) = @_;
+	
+	$self->{wanted_state} = undef;
```

Comment:
The default value of `wanted_state` is `undef`. And the value doesn't matter if this function returns zero anyway, I think?

How about:
```
if ($condition_code !~ /^[01]$/) {
    $self->{error} = "Value '$condition_code' should be '0' or '1'";
    return 0;
}

$self->{wanted_state} = $condition_code;

return 1;
```